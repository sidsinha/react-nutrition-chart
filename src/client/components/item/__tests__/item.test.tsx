import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import Item from './../item';

const ItemProps = {
  id: 1, 
  name: "Oreo", 
  calories: 319,
  fat: 23,
  carbs: 123,
  protein: 45,
  selected: true
}

afterEach(cleanup);

test('render 1 item', async() => {
    const { getByTestId } = render(<Item {...ItemProps} />);
    expect(screen.getByText(ItemProps.name)).toBeInTheDocument();
    expect(getByTestId("checkbox_nutrition")).toHaveAttribute("checked");
  });

  test('validate style', async() => {
    const { container } = render(<Item {...ItemProps} />);
    expect(container.firstChild).toHaveClass('flex pv3 justify-around bb b--moon-gray');
  });

  test('matches snapshot', async() => {
    const { container } = render(<Item {...ItemProps} />)
    expect(container.firstChild).toMatchSnapshot();
  });