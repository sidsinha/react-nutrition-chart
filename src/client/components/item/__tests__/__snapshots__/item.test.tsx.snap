// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="flex pv3 justify-around bb b--moon-gray bg-washed-red item"
>
  <div
    class="ml3"
    style="flex: 0.25;"
  >
    <input
      checked=""
      data-testid="checkbox_nutrition"
      id="item-Oreo-checkbox"
      name="nutrition"
      type="checkbox"
      value="1"
    />
  </div>
  <div
    class="item-name"
    style="flex: 1;"
  >
    Oreo
  </div>
  <div
    class="tc"
    style="flex: 0.5;"
  >
    319
  </div>
  <div
    class="tc"
    style="flex: 0.5;"
  >
    23
  </div>
  <div
    class="tc"
    style="flex: 0.5;"
  >
    123
  </div>
  <div
    class="tc"
    style="flex: 0.5;"
  >
    45
  </div>
</div>
`;
